library/src/animationManager.cxx

```diff
@@ -273,4 +272,27 @@ bool animationManager::LoadAtTime(double timeValue)
   }
   return true;
 }
+void animationManager::CycleAnimation()
+{
+  assert(this->Importer);
+  int numberOfAnimations = this->Importer->GetNumberOfAnimations();
+  if (numberOfAnimations <= 0)
+  {
+    return;
+  }
+  this->Importer->DisableAnimation(this->AnimationIndex);
+  this->AnimationIndex = (this->AnimationIndex + 1) % numberOfAnimations;
+  this->Importer->EnableAnimation(this->AnimationIndex);
+  this->LoadAtTime(0);
```

Comment:
what if the animation is currently playing ? Should we try to not reset the time ?

Reply:
In my opinion, it feels a lot cleaner to see the animation start from the beginning after each cycle. When it continues where it left off, it feels a bit awkward. The user would have to watch extra in order to see the entire animation play out again from the beginning. What do you think?